src/lib/src/auth/auth.component.ts

```diff
@@ -61,7 +61,20 @@ export class AuthComponent implements OnChanges, OnDestroy {
     // 1st load has no query, 2nd redirect-load does
     if (query) {
       let token = AuthParser.parseToken(query);
-      this.authService.setToken(token);
+      let error = AuthParser.parseError(query);
+      if (error) {
+        if (error == 'invalid_scope') {
+          // we don't want to use setError because this error is final.
+          // i.e. we don't want to prompt another login attempt, because
+          // the error is not with authentication, it's with authorization.
+          this.authService.setToken('AUTHORIZATION_FAIL');
```

Comment:
yup, we're definitely doing truthy token checks as a practice. Those shouldn't be too hard to find and update. Metrics only has a couple; Publish has a few more.

Reply:
I started down the road of throwing an error (see a98e15ea1abbdf3290a1c50deeb4a8180c7c8c02) because I think I feel your same ambivalence. It seems more JS-friendly to throw an error and catch it somewhere else and Do The Right Thing.

What I found was that it was more invasive to the code base as it exists currently in the various apps, at least insofar as I could understand, to try and catch errors on the token acquisition process. Everything currently assumes that a failure to get the token is not an error per se (which aborts/interrupts the observed AuthService), but a process to be re-tried. 

This is likely my own ignorance of how best to apply the error pattern to asych programming. I tried several different approaches, all of which proved more complicated than this PR.

I have started https://github.com/PRX/publish.prx.org/pull/507 to see how the truthiness of the token affects an app.